sql_generators/mobile_kpi_support_metrics/__init__.py

```diff
@@ -27,70 +28,118 @@
     "engagement.view.sql",
 )
 
-INSTALL_SOURCE = [
-    {
-        "name": "install_source",
-        "type": "STRING",
-        "description": "The source of a profile installation.",
-    },
-]
-
-ADJUST_FIELDS = [
-    {
-        "name": "adjust_ad_group",
-        "type": "STRING",
-        "description": "Adjust Ad Group the profile is attributed to.",
-    },
-    {
-        "name": "adjust_campaign",
-        "type": "STRING",
-        "description": "Adjust Campaign the profile is attributed to.",
-    },
-    {
-        "name": "adjust_creative",
-        "type": "STRING",
-        "description": "Adjust Creative the profile is attributed to.",
-    },
-    {
-        "name": "adjust_network",
-        "type": "STRING",
-        "description": "Adjust Network the profile is attributed to.",
-    },
-]
-
-FENIX_ATTRIBUTION_FIELDS = [
-    {
-        "name": "play_store_attribution_campaign",
-        "type": "STRING",
-        "description": "Play store campaign the profile is attributed to.",
-    },
-    {
-        "name": "play_store_attribution_medium",
-        "type": "STRING",
-        "description": "Play store Medium the profile is attributed to.",
-    },
-    {
-        "name": "play_store_attribution_source",
-        "type": "STRING",
-        "description": "Play store source the profile is attributed to.",
-    },
-    {
-        "name": "meta_attribution_app",
-        "type": "STRING",
-        "description": "Facebook app linked to paid marketing.",
-    },
-    *INSTALL_SOURCE,
-    *ADJUST_FIELDS,
-]
-
-FIREFOX_IOS_ATTRIBUTION_FIELDS = [
-    {
-        "name": "is_suspicious_device_client",
-        "type": "BOOLEAN",
-        "description": "Flag to identify suspicious device users, see bug-1846554 for more info.",
-    },
-    *ADJUST_FIELDS,
-]
+
+class Pings(Enum):
+    """TODO."""
+
+    metrics: str = "metrics"
+    first_session: str = "first_session"
+    baseline: str = "baseline"
+
+
+@dataclass
+class AttributionFieldGroup:
+    """TODO."""
+
+    name: str
+    source_pings: list[Pings]
+    fields: list[dict[str, str]]
+
+
+class AttributionFields:
+    """TODO."""
+
+    install_source = AttributionFieldGroup(
+        name="install_source",
+        source_pings=[Pings.metrics],
+        fields=[
+            {
+                "name": "install_source",
+                "type": "STRING",
+                "description": "The source of a profile installation.",
+            },
+        ],
+    )
+    adjust = AttributionFieldGroup(
+        name="adjust",
+        source_pings=[Pings.metrics, Pings.first_session],
+        fields=[
+            {
+                "name": "adjust_ad_group",
+                "type": "STRING",
+                "description": "Adjust Ad Group the profile is attributed to.",
```

Comment:
As the official attribution table, how about adding a more explicit description of each data point? This is the [adjust documentation](https://help.adjust.com/en/article/links#campaign-structure-parameters.) where definitions can be obtained. This link is available in the metadata of `firefox_android_clients` however it would have better visibility if it's in the description.

Reply:
I believe this is outside of the scope of this PR as this PR only makes small adjustments to the data structures to make it easier to add new_profiles and attribution tables and we're trying to keep as much of the configuration the same as was. If we think the descriptions are inaccurate or need to be updated we can do that as a separate change aimed at doing this.